source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_raw_basis_data_df.py

```diff
@@ -0,0 +1,466 @@
+        # when_connection_state_is_not_E22_Or_E23__returns_no period
+        (
+            {"start": june_1th, "end": june_10th},
+            [
+                {"FromDate": june_1th, "ToDate": june_3th, "EnergySupplierId": "1"},
+                {"FromDate": june_3th, "ToDate": june_6th, "EnergySupplierId": "2"},
+                {"FromDate": june_6th, "ToDate": june_10th, "EnergySupplierId": "3"},
+            ],
+            [
+                # not_used
+                {"FromDate": june_2th, "ToDate": june_3th, "ConnectionState": "D01"},
+                # closed_down
+                {"FromDate": june_3th, "ToDate": june_4th, "ConnectionState": "D02"},
+                # new
+                {"FromDate": june_4th, "ToDate": june_5th, "ConnectionState": "D03"},
+                # connected
+                {"FromDate": june_5th, "ToDate": june_6th, "ConnectionState": "E22"},
+                # disconnected TODO: LRN den vi have inkluderet.
+                {"FromDate": june_6th, "ToDate": june_7th, "ConnectionState": "E23"},
+            ],
+            [
+                {"EDate": june_5th, "toEDate": june_6th, "ESGln": "2", "SetM": "D01"},
+            ],
+        ),
+        # when_periods_connection_state_is_closed_down_period_is_not_includded
+    ],
+)
+def test__returns_expected_periods(
+    batch_period,
+    market_roles_periods,
+    metering_points_periods,
+    expected_periods,
+    grid_area_df,
+    market_roles_period_df_factory,
+    metering_points_periods_df_factory,
+):
+
+    metering_points_periods_df = metering_points_periods_df_factory(
+        periods=metering_points_periods
+    )
+    market_roles_periods_df = market_roles_period_df_factory(
+        periods=market_roles_periods
+    )
+
+    raw_master_basis_data = _get_master_basis_data_df(
+        metering_points_periods_df,
+        market_roles_periods_df,
+        grid_area_df,
+        batch_period["start"],
+        batch_period["end"],
+    )
+    raw_master_basis_data.show()
+    assert raw_master_basis_data.count() == len(expected_periods)
+    for expected_period in expected_periods:
+        period = raw_master_basis_data.where(
+            (col("EnergySupplierGln") == expected_period["ESGln"])
+            & (col("EffectiveDate") == expected_period["EDate"])
+            & (col("toEffectiveDate") == expected_period["toEDate"])
+            & (col("SettlementMethod") == expected_period["SetM"])
+        )
+
+        assert period.count() == 1
+
+
+# these test is done in test__returns_expected_periods. So the can they be removed?
+# def test__when_connection_state_is_E22__returns_metering_point_period():
+#     pass
+#
+#
+# def test__when_connection_state_is_not_E22__returns_none():
+#     pass
+#
+# def test__when_in_period__returns_metering_point_period():
+#     # mp_start, mp_end):
+#     # <= and >= combinations
+#     # period_start = xx
+#     # period_end = xx
+#     pass
+
+
+def test__when_type_is_E18__returns_metering_point_period(
```

Comment:
Second thought. I'd prefer
```suggestion
def test__when_type_is_production__returns_metering_point_period(
```
as I think that's lowering the entry level for developers working with the source code. Memorizing/learning the codes can then be deferred to on a need to know basis. E.g. when investigating actual storage data.